appcenter-data/ios/AppCenterReactNativeData/AppCenterReactNativeData.m

```diff
@@ -50,24 +61,102 @@ + (void)register {
-                 dict[kMSId] = document.documentId;
-                 dict[kMSETag] = document.eTag;
-                 dict[kMSLastUpdatedDate] = @([document.lastUpdatedDate timeIntervalSince1970] * 1000);
-                 dict[kMSIsFromDeviceCache] = [NSNumber numberWithBool:document.fromDeviceCache];
-                 return resolve(dict);
-             }];
+                     partition:partition
+                   readOptions:readOptions
+             completionHandler:[self dataCompletionHandler:@"Failed read" reject:reject resolve:resolve]];
+}
+
+RCT_EXPORT_METHOD(create:(NSString *)documentID
+                  partition:(NSString *)partition
+                  document: (NSDictionary *)documentMap
+                  writeOptions:(NSDictionary *)writeOptionsMap
+                  resolver:(RCTPromiseResolveBlock)resolve
+                  rejecter:(RCTPromiseRejectBlock)reject){
+    MSWriteOptions * writeOptions = [self getWriteOptions:writeOptionsMap];
+    MSDictionaryDocument *document = [[MSDictionaryDocument alloc] initFromDictionary:documentMap];
+    [MSData createDocumentWithID:documentID
+                        document:document
+                       partition:partition
+                    writeOptions:writeOptions
+               completionHandler:[self dataCompletionHandler:@"Failed create" reject:reject resolve:resolve]];
+}
+
+RCT_EXPORT_METHOD(replace:(NSString *)documentID
+                  partition:(NSString *)partition
+                  document:(NSDictionary *)documentMap
+                  writeOptions:(NSDictionary *)writeOptionsMap
+                  resolver:(RCTPromiseResolveBlock)resolve
+                  rejecter:(RCTPromiseRejectBlock)reject){
+    MSWriteOptions * writeOptions = [self getWriteOptions:writeOptionsMap];
+    MSDictionaryDocument *document = [[MSDictionaryDocument alloc] initFromDictionary:documentMap];
+    [MSData replaceDocumentWithID:documentID
+                         document:document
+                        partition:partition
+                     writeOptions:writeOptions
+                completionHandler:[self dataCompletionHandler:@"Failed replace" reject:reject resolve:resolve]];
+}
+
+RCT_EXPORT_METHOD(remove:(NSString *)documentID
+                  partition:(NSString *)partition
+                  writeOptions:(NSDictionary* )writeOptionsMap
+                  resolver:(RCTPromiseResolveBlock)resolve
+                  rejecter:(RCTPromiseRejectBlock)reject) {
+    MSWriteOptions * writeOptions = [self getWriteOptions:writeOptionsMap];
+    [MSData deleteDocumentWithID:documentID
+                       partition:partition
+                    writeOptions:writeOptions
+               completionHandler:[self dataCompletionHandler:@"Failed remove" reject:reject resolve:resolve]];
+}
+
+- (void (^)(MSDocumentWrapper * _Nonnull))dataCompletionHandler:(NSString*)errorCode
+                                                         reject:(RCTPromiseRejectBlock)reject
```

Comment:
Please put `resolve` parameter before `reject` parameter to be consistent. The convention is put resolve before reject for Promises.

Reply:
Changed